src/libp2p/connection/established/single_stream.rs

```diff
@@ -258,12 +266,17 @@ where
                 .map_err(Error::Yamux)?;
             self.inner.yamux = yamux_decode.yamux;
 
-            // TODO: it is possible that the yamux reading is blocked on writing
+            // If bytes_read is 0 and detail is None, then Yamux can't do anything more. On the
```

Comment:
I don't see how this comment adds anything. it just duplicates the code basically

Reply:
It's true that it's repeating a piece of information of the Yamux API.
But IMO, the fact that we have to loop again if Yamux didn't process all the data or returned an event is a bit subtle (especially the "or" here, so for example if Yamux processed all the data and returned an event we have to loop again), and is worth repeating for clarity.